core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -1747,11 +1758,33 @@ final class Stream[+F[_], +O] private[fs2] (private[fs2] val underlying: Pull[F,
       case None    => Pull.output1(fallback)
     }.stream
 
-  /** Applies the specified pure function to each input and emits the result.
+  /** Applies the function to each input and emits the results.
+    *
+    * This combinator keeps chunk structure and order: each output chunk is the
+    * result of applying `ch.map(f)` to a chunk `ch` from the source.
+    *
+    * **Errors** If an error is raised when pulling from the source (`this`)
+    * stream, then the mapped stream rethrows that error. If the function `f`
+    * throws an Exception `e` when applied to an element `o`, then the entire
+    * chunk containing that element is discarded, and the stream fails with
+    * that exception `e`.
+    *
+    * **Interruption**: if an interruption occurs, then the mapped stream is
+    * cut between chunks: the last output chunk is still the image of a whole
+    * source chunk. Once an interrupted is detected, the stream stops pulling
+    * from the source chunk. However, an interruption can be detected after
+    * a chunk is pulled from the source but before it is mapped and emitted.
+    * When this happens, that chunk is dropped and lost.
+    *
+    * This function follows the Functor laws.
```

Comment:
I'd remove this note about functor laws - unless otherwise stated, all instances should be assumed lawful.